test/units/testsuite-71.sh

```diff
@@ -89,10 +89,47 @@ test_chassis() {
     fi
 }
 
+restore_sysfs_dmi() {
+    umount /sys/class/dmi/id
+    rm -rf /run/systemd/system/systemd-hostnamed.service.d
+    systemctl daemon-reload
+    systemctl restart systemd-hostnamed
+}
+
+test_firmware_date() {
+    trap restore_sysfs_dmi RETURN
+
+    # Ignore /sys being mounted as tmpfs
+    mkdir /run/systemd/system/systemd-hostnamed.service.d/
+    cat > /run/systemd/system/systemd-hostnamed.service.d/override.conf <<- EOF
+[Service]
+Environment="SYSTEMD_DEVICE_VERIFY_SYSFS=0"
+EOF
+    systemctl daemon-reload
+
+    # mkdir for s390x
+    mkdir -p /sys/class/dmi/id
+    mount -t tmpfs none /sys/class/dmi/id
+    echo '1' > /sys/class/dmi/id/uevent
+
+    echo '01/01/2000' > /sys/class/dmi/id/bios_date
+    systemctl restart systemd-hostnamed
```

Comment:
Also here. stopping hostnamed is enough, as `hostnamectl` below will start hostnamed.